addons/crm/models/crm_lead.py

```diff
@@ -1957,9 +1957,9 @@ def _pls_get_naive_bayes_probabilities(self, batch_mode=False):
                 if field == 'stage_id' and value in won_stage_ids:
                     won_leads.add(lead_id)
                 leads_fields.add(field)
-
+        leads_fields = sorted(leads_fields)
         # get all variable related records from frequency table, no matter the team_id
-        frequencies = self.env['crm.lead.scoring.frequency'].search([('variable', 'in', list(leads_fields))], order="team_id asc")
+        frequencies = self.env['crm.lead.scoring.frequency'].search([('variable', 'in', list(leads_fields))], order="team_id asc, id")
```

Comment:
Shouldn't we set this as model order (seems it is missing) instead ?

Reply:
And ``crm.lead.scoring.frequency.field`` has no order non plus

Reply:
True, setting the order on the model may be a good idea, but I'm not sure this is wanted. My goal is to do transparent fix, not to  change the behavior without knowing the business needs :) 
If a model has no order the default order is 'id' so there is no problem in this case. 
A potential problem is to set on a model _order = 'non_unique_field', in this case a falback on id is a good idea.